glusterd2/pmap/registry.go

```diff
@@ -107,6 +107,22 @@ func (r *pmapRegistry) SearchByBrickPath(brickpath string) (int, error) {
 	return -1, fmt.Errorf("SearchByBrickPath: port for brick %s not found", brickpath)
 }
 
+// NumOfBricksOnPort returns number of bricks attached to a port. Called during
+// brick multiplexing when there is a max-brick-per-process constraint
+func (r *pmapRegistry) NumOfBricksOnPort(port int) (int, error) {
```

Comment:
`pmapRegistry` is in-memory, need to test this in different scenarios.
as @atinmu said what happens when glusterd2 restarts?

Reply:
@atinmu  @Madhu-1  If we are not using pmapRegistry for checking this and opt for mantaining list of bricks attached to every brick instance then we will need to store this data in etcd store for GD2 restarts.

Reply:
Also, the original brick-multiplexing (where max-bricks-per-process is not set) is implemented based on pmap as well.